### DISCUSSION TITLE
signMessage on metamask app sometimes ask me to switch network.

### DISCUSSION THREAD

**Discussion by joshbenandiel** [WalletConnect/walletconnect-monorepo#3523] [Developer Support] [answered]
After calling sign message, it redirects me to metamask app then switch network opens. It's either "cancel" or "confirm" options after clicking one of those it redirects me back to app and i tried to manually switch to metamask app sign message won't popup. Is there a way to get the data wether "cancel" or "confirm" is clicked? Thanks for the help btw I don't have default chain in `<Web3Modal />`

**Comment by glitch-txs** [comment]
Hi! Thanks for reporting, kindly could you share a [MRE](https://github.com/orgs/WalletConnect/discussions/3457)?

**Reply by joshbenandiel** [reply]
Hi, i'm using `useSignMessage` wagmi then platform was mobile web on Android. Wallet used `MetaMask` app

To reproduce:
Close metamask app then try to use to `signMessage` in `useSignMessage` sometimes it was asking me to switch network. It's like when metamask app is opened for the first time. When i clicked "cancel" next sign message call won't ask me to switch network it works fine. Sign message popup will show at this point

**Reply by joshbenandiel** [reply]
It works perfectly fine when i'm using injected connector even though switch network is being asked sign message popup still shown

**Reply by joshbenandiel** [reply]
is there a way to disable network switching when signing a message?

**Reply by glitch-txs** [reply, marked as answer]
Hi this is on MetaMask mobile side, I think you should try reaching out to them. We'll take a look on this as well

**Reply by joshbenandiel** [reply]
Got it thanks!
